Lexis+.js

```diff
@@ -0,0 +1,267 @@
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-04-09 22:52:49"
+}
+
+/*
+  ***** BEGIN LICENSE BLOCK *****
+
+  Copyright © 2023 Brandon Fahrenfort
+
+  This file is part of Zotero.
+
+  Zotero is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Zotero is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+  ***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, _url) {
+	if (doc.title.match(/.*results.*/)) {
+		return "multiple";
+	}
+	else if (doc.title.match(/[a-zA-Z. ]+\s§\s\d+/)
+	|| doc.title.match(/act/i)
+	|| doc.title.match(/p\.l\./i)) { // Match: ... Tex. Bus. & Com. Code § 26.01 ...
+		return "statute";
+	}
+	else if (doc.title.match(/\d+\s[a-zA-Z0-9. ]+\s\d+/)) { // Match: ... 5 U.S. 137 ...
+		return "case";
+	}
+	// TODO secondary sources
+
+	return false;
+}
+
+function getSearchResults(doc, url) {
+	var items = {};
+	var nextTitle;
+
+	if (detectWeb(doc, url) == "multiple") {
+		// TODO check what type of element it is (currently only working for 'cases' searches)
+		var titles = doc.querySelectorAll('a.titleLink');
+		var dates = doc.querySelectorAll('span.metaDataItem'); // Not technically only dates, but that's all I use it for atm
+		var nextDate;
+		var dateOffset = 1;
+	
+		// dates[0] is first court name
+		nextDate = dates[dateOffset];
+		dateOffset += 3;
+		// dates[2] is first citation
+		for (var i = 0; i < titles.length; i++) {
+			nextTitle = titles[i];
+			items[nextTitle.href] = nextTitle.textContent + "(" + nextDate.textContent + ")";
+
+			// dates[0] is court name
+			nextDate = dates[dateOffset];
+
+			// dates[2] is a citation
+		}
+
+		return items;
+	}
+
+	return false;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) == 'multiple') {
+		let items = await Zotero.selectItems(getSearchResults(doc, url));
+		if (!items) return;
+		for (let url of Object.keys(items)) {
+			await scrape(await requestDocument(url));
+		}
+	}
+	else {
+		await scrape(doc, url);
+	}
+}
+
+async function scrape(doc, url) {
+	var title = text(doc, 'h1#SS_DocumentTitle');
+
+	if (detectWeb(doc, url) == "case") {
+		var newCase = new Zotero.Item("case");
+		//newCase.url = doc.location.href; // Disabled for style reasons
+
+		newCase.title = title;
+
+		newCase.notes.push({note: "Snapshot: " + newCase.title + doc.getElementById('document-content').innerHTML});
+
+		var citation = text(doc, 'span.active-reporter');
```

Comment:
```suggestion
		let citation = text(doc, 'span.active-reporter');
```
better for temporary objects. Also means this won't fail linting if you use `citation` again for a different item type